src/widget-loader.ts

```diff
@@ -105,13 +105,20 @@ interface CustomTemplate<C> {
   template?: Template<C>
 }
 
+interface WidgetConfig {
+  style: Partial<Style>
+  expandedTile: Partial<ExpandedTileOptions>
+  inlineTile: Partial<InlineTileOptions>
+}
+
 type Templates<C> = Record<string, CustomTemplate<C>>
 
 export interface MyWidgetSettings<C> {
```

Comment:
Instead of this, you can do something like

```js
export type EnforcedWidgetSettings<C> = {
  features: Features
  callbacks: Callbacks
  extensions: Extensions
  templates: Partial<Templates<C>>
  config: Partial<WidgetConfig>
} 
```

For all optional type

```js
type MyWidgetSettings<C> = Partial<EnforcedWidgetSettings<C>> & { font?: string } 
```

something like that

Reply:
Brilliant, good idea. Thanks!